opentelemetry-sdk/src/opentelemetry/sdk/_shared_internal/__init__.py

```diff
@@ -39,6 +39,26 @@
 from opentelemetry.util._once import Once
 
 
+class DuplicateFilter(logging.Filter):
```

Comment:
Isn't context based suppression the right way to prevent self-logging?
https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/export/__init__.py#L546